### PR TITLE
Fix bug introduced by

### PR DESCRIPTION
**Pull request by vivekmenezes**
https://github.com/cockroachdb/cockroach/pull/1640

### CODE REVIEW

File: storage/range.go [hunk 1]
```diff
@@ -964,19 +964,18 @@ func (r *Range) getLeaseForGossip(ctx context.Context) (hasLease bool, err error
 	hasLease = true
 	if !r.rm.Stopper().RunTask(func() {
 		timestamp := r.rm.Clock().Now()
-
 		// Check for or obtain the lease, if none active.
 		err = r.redirectOnOrAcquireLeaderLease(tracer.FromCtx(ctx), timestamp)
 		if err != nil {
+			hasLease = false
```
Comment (bdarnell) [inline storage/range.go]: Instead of setting hasLease to false here, can we default it to false and change it to true only when err is nil?

Comment (vivekmenezes) [inline storage/range.go]: done

**Comment by vivekmenezes** [comment]
Took the liberty to merge this in because I had to leave for the evening. Thanks

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**